src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -254,7 +254,10 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                 return false;
             }
 
-            return true;
+            // Auto-properties (readonly, get, set) marked with an attribute have a block context callback while they don't
+            // if there is no attribute (see https://github.com/dotnet/roslyn/issues/46132).
+            // We consider that auto-property have the intent to always be instance members so we want to workaround this issue.
+            return !methodSymbol.IsAutoPropertyAccessor();
```

Comment:
```suggestion
            // We consider that auto-property have the intent to always be instance members, so we explicitly skip them.
            return !methodSymbol.IsAutoPropertyAccessor();
```

Reply:
Isn't it better to keep track of the ticket? Because depending on how this is implemented later on, it's possible we will no longer have the callback (or on the opposite we always have the callback).

Reply:
Hmm do you still feel any operation tree shape based approach can be better then current symbol based detection approach? IMO, even if the callback was made, it seems more reasonable to not rely on operation tree shape for such detection. I'll let you make a call, feel free to ignore this suggestion.